Duplicate SPN performance

mmorrison08 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [code style](https://www.asbuiltreport.com/about/contributing/#use-a-consistent-coding-style) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [contributing](https://www.asbuiltreport.com/about/contributing/) documentation.

rebelinux commented:
What is the intention of the merge? Any issues or errors that you are experiencing?

Let me know if you need help!

mmorrison08 commented:
Sorry for the merge, did not mean to submit it.

There are a couple problems that are in the module that I corrected (first that comes to mind is the zero group member report). I'm also taking a deeper look into performance enhancements throughout. In some larger environments I've ran the script against, some sections were taking 4-5 hours but I've gotten down to 30 minutes or so through various optimizations. I've also been adding a lot more verbose outputs throughout the script so it is more obvious where some sections are running for a long time.

For now I'm just working off of a fork with those corrections, but I will circle back and at least do requests against the problem areas I fixed as well as some of the optimizations. As my fork currently is, I think there are too many changes than you would want in one go.

rebelinux commented:
> Sorry for the merge, did not mean to submit it.
> 
> There are a couple problems that are in the module that I corrected (first that comes to mind is the zero group member report). I'm also taking a deeper look into performance enhancements throughout. In some larger environments I've ran the script against, some sections were taking 4-5 hours but I've gotten down to 30 minutes or so through various optimizations. I've also been adding a lot more verbose outputs throughout the script so it is more obvious where some sections are running for a long time.
> 
> For now I'm just working off of a fork with those corrections, but I will circle back and at least do requests against the problem areas I fixed as well as some of the optimizations. As my fork currently is, I think there are too many changes than you would want in one go.

Understood. Thanks for letting me know.

I have been thinking about restructuring the report. If you have any ideas or if you think more sections should be added, don't hesitate to let me know!

PS: Remember to synchronize your repository before performing the Pull Request.
